### PR TITLE
JavaScript bundling with WP-Scripts and Webpack #467

### PR DESCRIPTION
**Pull request by erikyo**
## Summary
This pull request enables the use of wp-scripts in order to compile ES to JS. This intrinsically brings many improvements, among them code minification, transpilation to the most used version of js and an increased compatibility with various devices thanks to browserlists. No stylesheet are available at the moment but in the future it might be useful to know that we can now use also scss

## Relevant technical choices
- 2 script were added in package.json, build-js and start-js and are used to do exactly what is documented [here](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#setup)
- The file `webpack.config.js` was added with a funtion used to "enqueue" scripts, actually is used as suggested by @felixarntz to bundle fallback.js but any other script could be added in this way:

```javascript
const ref= addModule( 'myfile.js', 'modules/moduleType/moduleName/' );
```

and then simply add to module.exports an array with all the needed modules comma separated, like below
`module.exports = [first, second, etc]`

- Changed the `modules/images/webp-uploads/load.php` path since must now point to the minified version inside the dist folder, but I haven't published that one, so if you need it will be a good opportunity for a double check 😉 

## Checklist

- [ ] PR has either `[Focus]` or `Infrastructure` label.
- [ ] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

### CODE REVIEW

**Comment by mxbclang** [comment]
PR for #467

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@octokit/rest": "16.26.0",
     "@wordpress/env": "^4.2.2",
     "@wordpress/scripts": "^19.0",
+    "@wordpress/browserslist-config": "^4.1.2",
```
Comment (ntwb) [inline package.json]: This is included in the `@wordpress/scripts` package, so no need to add it separately

https://github.com/WordPress/gutenberg/blob/83a5df739dc904bc106a36002674aa7c68130aee/packages/scripts/package.json#L39

Comment (erikyo) [reply package.json]: yep! Thank you, good catch! fixed!

**Comment by erikyo** [comment]
closing because outdated!

**Pull request closed without merging**